cmd/export.go

```diff
@@ -59,11 +60,29 @@ func exportDashboardsCmd(cmd *cobra.Command, args []string) error {
 		opts = append(opts, kibana.TLSSkipVerify())
 	}
 
+	allowSnapshot, _ := cmd.Flags().GetBool(cobraext.AllowSnapshotFlagName)
+	if err != nil {
+		return cobraext.FlagParsingError(err, cobraext.AllowSnapshotFlagName)
+	}
+
 	kibanaClient, err := kibana.NewClient(opts...)
 	if err != nil {
 		return errors.Wrap(err, "can't create Kibana client")
 	}
 
+	kibanaVersion, isSnapshot, err := kibanaClient.Version()
+	if err != nil {
+		return errors.Wrap(err, "can't get Kibana status information")
+	}
+
+	if isSnapshot {
+		message := fmt.Sprintf("Exporting dashboards from a Elastic stack SNAPSHOT version (%s) is discouraged. It could lead to invalid dashboards (e.g. using features that could be not released or they could changed)", kibanaVersion)
+		if !allowSnapshot {
+			return errors.Errorf("using Elastic stack SNAPSHOT version: %s. %s", kibanaVersion, message)
```

Comment:
Nit. The version is already in the message. Maybe we can add here a mention about the flag.
```suggestion
			return errors.Errorf("%s. --%s flag can be used to ignore this error", message, cobraext.AllowSnapshotFlagName)
```